en/Building_a_Simple_Engine/Engine_Architecture/05_rendering_pipeline.adoc

```diff
@@ -0,0 +1,1515 @@
+
+                     // Execute screen-space lighting calculations
+                     cmd.beginRendering(renderingInfo);                              // Begin lighting pass
+
+                     // Bind lighting pipeline and draw full-screen quad
+                     // Fragment shader reads G-Buffer textures and computes lighting for each pixel
+                     // All scene lights are processed in a single screen-space pass
+                     // ... (lighting calculation implementation would go here)
+
+                     cmd.endRendering();                                             // Complete lighting calculations
+                 });
+
+    // Compile the complete rendergraph for execution
+    // This analyzes dependencies and generates optimal execution plan
+    graph.Compile();
+}
+----
+
+The lighting pass represents the core advantage of deferred rendering: decoupling lighting complexity from geometric complexity. By processing lighting in screen space, the cost becomes proportional to screen resolution rather than scene complexity, enabling efficient handling of scenes with many lights or complex lighting models.
+
+The single render target configuration reflects the unified nature of the lighting pass, where all lighting contributions are accumulated into the final color buffer. This approach enables advanced lighting techniques like physically-based rendering or global illumination algorithms that would be prohibitively expensive in forward rendering scenarios with complex geometry.
+
+==== Best Practices for Rendergraphs and Synchronization
+
+1. *Minimize Synchronization*: Use the rendergraph to minimize the number of synchronization points.
+2. *Batch Similar Operations*: Group similar operations together to reduce state changes.
+3. *Use Appropriate Access Flags*: Be specific about which access types you need to synchronize.
+4. *Avoid Redundant Barriers*: Let the rendergraph eliminate redundant barriers.
+5. *Consider Memory Aliasing*: Use the rendergraph's memory aliasing capabilities to reduce memory usage.
+6. *Profile and Optimize*: Use GPU profiling tools to identify synchronization bottlenecks.
+7. *Handle Platform Differences*: Different GPUs may have different synchronization requirements.
+
+[source,cpp]
+----
+// Forward declarations
+class RenderPass;
+class RenderTarget;
+
+// Render pass manager
+class RenderPassManager {
+private:
+    std::unordered_map<std::string, std::unique_ptr<RenderPass>> renderPasses;
+    std::vector<RenderPass*> sortedPasses;
+    bool dirty = true;
+
+public:
+    template<typename T, typename... Args>
+    T* AddRenderPass(const std::string& name, Args&&... args) {
+        static_assert(std::is_base_of<RenderPass, T>::value, "T must derive from RenderPass");
+
+        auto it = renderPasses.find(name);
+        if (it != renderPasses.end()) {
+            return dynamic_cast<T*>(it->second.get());
```

Comment:
Again: how does the user know that all of its arguments provided for a new `RenderPass` are ignored?